src/deprecated.jl

```diff
@@ -0,0 +1,15 @@
+using Base: depwarn
+
+export recordproperty
+
+function recordproperty(name::String, val)
+    depwarn("`recordproperty` is deprecated, use `record_testset_property` instead.", :recordproperty)
+    record_testset_property(name, val)
+    return nothing
+end
+
+function properties(ts::AbstractTestSet)
+    depwarn("`$properties` is deprecated, use `$testset_properties` instead.", :properties)
+    p = testset_properties(ts)
+    return isnothing(p) ? nothing : Dict(p)
+end
```

Comment:
Probably want to make this PR breaking and avoid the deprecations as we no longer allow properties to be overwritten

Reply:
Was that a feature before though, or a bug?

Reply:
I guess it was a feature, as it was a documented thing you were warned against doing but were allowed to do.
So it probably should indeed be a breaking change.

When we merge this shall we take the jump straight to 1.0.0?
I do not like having a 0.x.y package

Reply:
I'm good with jumping to 1.0.0